src/config/config.ts

```diff
@@ -1,193 +1,110 @@
+    .getProperties()
 }
 
-/**
- * FormSG SDK mode.
- * Needed due to the SDK having multiple modes - staging, production,
- * development, and test, whilst the backend only has 2 modes (prod and dev).
- *
- * The SDK modes are required to be set properly so that the correct public keys
- * are used to sign, encrypt, or decrypt data that is passed into the SDK.
- */
-const formsgSdkMode = process.env.FORMSG_SDK_MODE as PackageMode
-if (
-  !formsgSdkMode ||
-  !['staging', 'production', 'development', 'test'].includes(formsgSdkMode)
-) {
-  throw new Error(
-    'FORMSG_SDK_MODE not found or invalid. Please specify one of: "staging" | "production" | "development" | "test" in the environment variable.',
-  )
+// Construct bucket URLs depending on node environment
+// If in development env, endpoint communicates with localstack, a fully
+// functional local AWS cloud stack for hosting images/logos/attachments.
+// Else, the environment variables to instantiate S3 are used.
+
+const awsEndpoint = isDev
+  ? defaults.aws.endpoint
+  : `https://s3.${basicVars.awsConfig.region}.amazonaws.com` // NOTE NO TRAILING / AT THE END OF THIS URL!
+
+// Perform validation before accessing s3 Bucket Urls
+const s3BucketUrlSchema = loadS3BucketUrlSchema(isDev)
+const s3BucketUrlVars = convict(s3BucketUrlSchema)
+  .load({
+    logoBucketUrl: `${awsEndpoint}/${basicVars.awsConfig.logoS3Bucket}`,
+    imageBucketUrl: `${awsEndpoint}/${basicVars.awsConfig.imageS3Bucket}`,
+    // NOTE THE TRAILING / AT THE END OF THIS URL! This is only for attachments!
+    attachmentBucketUrl: `${awsEndpoint}/${basicVars.awsConfig.attachmentS3Bucket}/`,
+  })
+  .validate({ allowed: 'strict' })
+  .getProperties()
+
+const s3 = new aws.S3({
+  region: basicVars.awsConfig.region,
+  // Unset and use default if not in development mode
+  // Endpoint and path style overrides are needed only in development mode for
+  // localstack to work.
+  endpoint: isDev ? defaults.aws.endpoint : undefined,
+  s3ForcePathStyle: isDev ? true : undefined,
+})
+
+const awsConfig: AwsConfig = {
+  ...s3BucketUrlVars,
+  ...basicVars.awsConfig,
+  s3,
 }
 
-// Optional environment variables
-/**
- * Name of CloudWatch log group to store short-term logs. Log streams are
- * separated by date.
- */
-const customCloudWatchGroup = process.env.CUSTOM_CLOUDWATCH_LOG_GROUP
-
-/**
- * Load env variable with General Maintenance banner text.
- */
-const isGeneralMaintenance = process.env.IS_GENERAL_MAINTENANCE
-
-/**
- * The banner message on login page. Allows for HTML.
- */
-const isLoginBanner = process.env.IS_LOGIN_BANNER
-
-/**
- * The banner message to show on all pages. Allows for HTML. Will supersede
- * all other banner content if it exists.
- */
-const siteBannerContent = process.env.SITE_BANNER_CONTENT
-
-/**
- * The banner message to show on on admin pages. Allows for HTML.
- */
-const adminBannerContent = process.env.ADMIN_BANNER_CONTENT
+const configLogger = createLoggerWithLabel('config')
```

Comment:
should move this to the top of the file. also there will be merge conflicts with karrui's logging PR - this will have to be changed to `createLoggerWithLabel(module)`.

Reply:
noted. will rebase once that is merged in